pkg/ingester/metrics_test.go

```diff
@@ -476,7 +478,8 @@ func TestTSDBMetricsWithRemoval(t *testing.T) {
 
 			# HELP cortex_ingester_tsdb_out_of_order_samples_appended_total Total number of out-of-order samples appended.
 			# TYPE cortex_ingester_tsdb_out_of_order_samples_appended_total counter
-			cortex_ingester_tsdb_out_of_order_samples_appended_total 9
```

Comment:
I don't understand. Before it was 9, after this change the total is 6. What's happening?

Reply:
Oh it's because an user is removed.

Reply:
Before the counter was global, now it has the user label. The test is populating metrics for 3 tsdb's and then removing the registry of the third tsdb
https://github.com/grafana/mimir/blob/9c842e09cd3422a3e2b891c3befb25c4837ba518/pkg/ingester/metrics_test.go#L264-L265

This is why `cortex_ingester_tsdb_out_of_order_samples_appended_total{user="user2"} 3` is not part of the metrics output and why now its 6 and not 9.